src/ocean/core/Optional.d

```diff
@@ -0,0 +1,210 @@
+/******************************************************************************
+
+    Wraps any type in a struct that also contains boolean field indicating if
+    value is in defined state.
+
+    If T is a value type, `Optional!(T)` is value type too.
```

Comment:
Missing `Params:`. This is a topic that I've been unsure about before, with a module header comment block being `/// ditto`ed onto an aggregate. Seems weird to have `Params:` in the header block, but it seems weird to _not_ have it on the aggregate's comment block.

Reply:
I considered same choice and decided `Params` in the module is much weirder.

Reply:
Yeah, for a simple case like this with just one template arg, I think this seems fine.